patches/api/0333-Added-getHostname-to-AsyncPlayerPreLoginEvent.patch

```diff
@@ -0,0 +1,64 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MCMDEV <john-m.1@gmx.de>
+Date: Fri, 24 Sep 2021 17:59:23 +0200
+Subject: [PATCH] Added getHostname to AsyncPlayerPreLoginEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+index 694a81769076ea58aae9f14f076ab80c9952c957..e6a0b2c9938ae59b6e88aa9128b6966ece02942a 100644
+--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
++++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+@@ -22,6 +22,7 @@ public class AsyncPlayerPreLoginEvent extends Event {
+     private final InetAddress ipAddress;
+     private final InetAddress rawAddress; // Paper
+     //private UUID uniqueId; // Paper - Not used anymore
++    private final String hostname; // Paper
+ 
+     @Deprecated
+     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
+@@ -66,7 +67,15 @@ public class AsyncPlayerPreLoginEvent extends Event {
+         this(name, ipAddress, ipAddress, uniqueId, profile);
+     }
+ 
++    // Paper - Add hostname
++    @Deprecated
+     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final InetAddress rawAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile) {
++        // Paper start - Add hostname
++        this(name, ipAddress, rawAddress, uniqueId, profile, "");
++    }
++
++    public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final InetAddress rawAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile, String hostname) {
++        // Paper end - Add hostname
+         super(true);
+         this.profile = profile;
+         // Paper end
+@@ -76,7 +85,9 @@ public class AsyncPlayerPreLoginEvent extends Event {
+         this.ipAddress = ipAddress;
+         this.rawAddress = rawAddress; // Paper
+         //this.uniqueId = uniqueId; // Paper - Not used anymore
++        this.hostname = hostname; // Paper - Add hostname
+     }
++    // Paper end
```

Comment:
Useless comment

Reply:
The comment in line 41?

Reply:
Yes; the previous paper start comment is already closed (L31), and the change here is only a one line (no start comment needed)